### DISCUSSION TITLE
How to switch (change) database connection dynamically in Csla Business objects

### DISCUSSION THREAD

**Discussion by Batzangia** [MarimerLLC/csla#2004] [Questions] [answered]
Hello everyone,

We are developing Csla domains. Now days we need to expose Csla domains as generic Api. 
Any way to change Csla ConnectionManager connectionString depending on Api request? (How to inject Csla ConnectionManager from Api Controllers  via Csla objects?)

Regards

**Comment by hurcane** [comment, marked as answer]
There are surely multiple ways to handle this. Our application allows the user to switch databases while using the app. We store the connection string in Csla.ApplicationContext.ClientContext.

**Reply by Batzangia** [reply]
Thanks @hurcane. 
We are storing the connection string in Csla.ApplicationContext.ClientContext according to your advice.
